consensus/core/src/block.rs

```diff
@@ -51,21 +64,45 @@ pub enum Block {
     V1(BlockV1),
 }
 
+impl Block {
+    /// Generate the genesis blocks for the latest Block version. The tuple contains (my_genesis_block, others_genesis_blocks).
+    /// The blocks are returned in authority index order.
+    pub(crate) fn genesis(context: Arc<Context>) -> (VerifiedBlock, Vec<VerifiedBlock>) {
```

Comment:
I think eventually we'll need things like the ProtocolConfig as well (ex if we migrate from BlockV1 to BlockV2) that's why I've considered using the Context directly. As for creating a vec of blocks, that's what I did intially but I found might self having to find my genesis block a few times and wanted to avoid that - that's why the split.